layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteNavigationRestService.java

```diff
@@ -192,6 +201,48 @@ public Response moveNode (@Parameter(description = "node id") @QueryParam("nodeI
     }
   }
 
+  @Path("{pageRef}")
+  @PUT
+  @RolesAllowed("users")
+  @Operation(summary = "update page access abd edit permission", method = "PUT", description = "update page access abd edit permission")
+  @ApiResponses(value = { @ApiResponse(responseCode = "204", description = "Request fulfilled"),
+          @ApiResponse(responseCode = "400", description = "Invalid query input"),
+          @ApiResponse(responseCode = "404", description = "Page not found"),
+          @ApiResponse(responseCode = "401", description = "Unauthorized operation"),
+          @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response updatePagePermissions(@Context
+  HttpServletRequest request,
+                                        @Parameter(description = "Page reference", required = true)
+                                        @PathParam("pageRef")
+                                        String pageRef,
+                                        @Parameter(description = "Page new edit permission", required = true)
+                                        @QueryParam("editPermission")
+                                        String editPermission,
+                                        @Parameter(description = "Page new access permissions", required = true)
+                                        @QueryParam("accessPermissions")
+                                        String accessPermissions) {
+    try {
+      if (pageRef == null || editPermission == null || editPermission.isBlank() ||  accessPermissions == null || accessPermissions.isEmpty()) {
+        return Response.status(Response.Status.BAD_REQUEST).entity("params are mandatory").build();
+      }
+      PageContext page = layoutService.getPageContext(PageKey.parse(pageRef));
+      if(page == null){
```

Comment:
```suggestion
      if (page == null) {
```

format the whole class